src/main/topLevelPages/myDashboard/myMemberships/MyMembershipsDialog.tsx

```diff
@@ -1,80 +1,69 @@
-import React from 'react';
-import { Trans, useTranslation } from 'react-i18next';
-import { DialogContent } from '@mui/material';
-import DialogWithGrid from '../../../../core/ui/dialog/DialogWithGrid';
-import DialogHeader from '../../../../core/ui/dialog/DialogHeader';
+import { ReactElement, JSXElementConstructor } from 'react';
+
+import { Trans } from 'react-i18next';
+import { SvgIconProps, DialogContent } from '@mui/material';
+
 import Gutters from '../../../../core/ui/grid/Gutters';
-import { useMyMembershipsQuery } from '../../../../core/apollo/generated/apollo-hooks';
 import { Caption } from '../../../../core/ui/typography';
-import RouterLink from '../../../../core/ui/link/RouterLink';
-import { SpaceIcon } from '../../../../domain/journey/space/icon/SpaceIcon';
-import { Visual } from '../../../../domain/common/visual/Visual';
-import {
-  AuthorizationPrivilege,
-  CommunityMembershipStatus,
-  CommunityRoleType,
-} from '../../../../core/apollo/generated/graphql-schema';
+import DialogHeader from '../../../../core/ui/dialog/DialogHeader';
+import DialogWithGrid from '../../../../core/ui/dialog/DialogWithGrid';
+
 import Loading from '../../../../core/ui/loading/Loading';
-import ExpandableSpaceTree from './ExpandableSpaceTree';
+import { ExpandableSpaceTree } from './ExpandableSpaceTree';
+import RouterLink from '../../../../core/ui/link/RouterLink';
+
 import { gutters } from '../../../../core/ui/grid/utils';
 import { useCreateSpaceLink } from '../useCreateSpaceLink/useCreateSpaceLink';
 
-interface MyJourneysDialogProps {
+type MyMembershipsDialogProps = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any[]; // TODO:p Find (create if doesn't exist) type for this.
+  title: string;
   open: boolean;
+  loading: boolean;
   onClose: () => void;
-}
-
-export interface MembershipProps {
-  id: string;
-  profile: {
-    url: string;
-    displayName: string;
-    tagline: string;
-    avatar?: Visual;
-    cardBanner?: Visual;
-  };
-  level: number;
-  authorization?: {
-    myPrivileges?: AuthorizationPrivilege[];
-  };
-  community?: {
-    myRoles?: CommunityRoleType[];
-    myMembershipStatus?: CommunityMembershipStatus;
-  };
-  childMemberships?: MembershipProps[];
-}
 
-const MyMembershipsDialog = ({ open, onClose }: MyJourneysDialogProps) => {
-  const { t } = useTranslation();
+  showFooterText?: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Icon?: ReactElement<SvgIconProps, string | JSXElementConstructor<any>>; // This any type must stay here.
```

Comment:
we're passing icons in many components, let's be strict with something like  ComponentType<SvgIconProps>